android/pytorch_android/src/main/java/org/pytorch/Tensor.java

```diff
@@ -143,31 +179,85 @@ public static long numElements(long[] dims) {
         "Tensor of type " + getClass().getSimpleName() + " cannot return data as float array.");
   }
 
-  public boolean isByteTensor() {
-    return TYPE_CODE_BYTE == getTypeCode();
-  }
-
-  public boolean isIntTensor() {
-    return TYPE_CODE_INT32 == getTypeCode();
+  public long[] getDataAsLongArray() {
+    throw new IllegalStateException(
+        "Tensor of type " + getClass().getSimpleName() + " cannot return data as float array.");
   }
 
-  public boolean isFloatTensor() {
-    return TYPE_CODE_FLOAT32 == getTypeCode();
+  public double[] getDataAsDoubleArray() {
+    throw new IllegalStateException(
+        "Tensor of type " + getClass().getSimpleName() + " cannot return data as double array.");
   }
 
-  abstract int getTypeCode();
-
   Buffer getRawDataBuffer() {
     throw new IllegalStateException(
         "Tensor of type " + getClass().getSimpleName() + " cannot " + "return raw data buffer.");
   }
 
-  private static String invalidIndexErrorMessage(int[] index, long dims[]) {
-    return String.format(
-        Locale.US,
-        "Invalid index %s for tensor dimensions %s",
-        Arrays.toString(index),
-        Arrays.toString(dims));
+  static class Tensor_byte extends Tensor {
+    private final ByteBuffer data;
+
+    private Tensor_byte(ByteBuffer data, long[] dims) {
+      super(dims);
+      this.data = data;
+    }
+
+    @Override
+    public int dtype() {
+      return DTYPE_BYTE;
+    }
+
+    @Override
+    Buffer getRawDataBuffer() {
+      return data;
+    }
+
+    @Override
+    public byte[] getDataAsByteArray() {
+      data.rewind();
+      byte[] arr = new byte[data.remaining()];
+      data.get(arr);
+      return arr;
+    }
+
+    @Override
+    public String toString() {
+      return String.format(
+          "Tensor_byte{shape:%s numel:%d}", Arrays.toString(shape), data.capacity());
```

Comment:
just for consistency with Python - can you print it with dtype=uint8? We mostly not use "byte" in modern api

```
>>> torch.zeros(5, dtype=torch.uint8)
tensor([0, 0, 0, 0, 0], dtype=torch.uint8)
```